Source/WebCore/css/typedom/transform/CSSMatrixComponent.cpp

```diff
@@ -155,7 +155,17 @@ void CSSMatrixComponent::serialize(StringBuilder& builder) const
 
 ExceptionOr<Ref<DOMMatrix>> CSSMatrixComponent::toMatrix()
 {
-    return { m_matrix.get() };
+    if (!is2D())
+        return { m_matrix.get() };
+
+    // Flatten to 2d.
+    return { DOMMatrix::create({
+        m_matrix->m11(),
+        m_matrix->m12(),
+        m_matrix->m21(),
+        m_matrix->m22(),
+        m_matrix->m41(),
+        m_matrix->m42() }, DOMMatrixReadOnly::Is2D::Yes) };
```

Comment:
I think it might be more explicit to use the 2D accessors (`a()`, `b()`, `c()`, `d()`, `e()` and `f()`) here.